cmd/gpu_plugin/README.md

```diff
@@ -205,6 +218,29 @@ GPU plugin does __not__ setup SR-IOV. It has to be configured by the cluster adm
 
 GPU plugin does however support provisioning Virtual Functions (VFs) to containers for a SR-IOV enabled GPU. When the plugin detects a GPU with SR-IOV VFs configured, it will only provision the VFs and leaves the PF device on the host.
 
+### KMD and UMD
+
+There are three different KMDs (Kernel Mode Driver) available: `i915 upstream`, `i915 backport` and `xe`:
+* `i915 upstream` is a vanilla driver that comes from the upstream kernel and is included in the common Linux distributions, like Ubuntu.
+* `i915 backport` is a [backported/out-of-tree driver](https://github.com/intel-gpu/intel-gpu-i915-backports/) and it has the freshest feature-set that will eventually make its way to the upstream kernel. Due to how upstreaming works with reviews and approvals, the feature-set might change in the upstreaming process.
+* `xe` is a new KMD that is intended to support future GPUs. It can work with the current newest GPUs (>=Tigerlake).
+
+For optimal performance, the KMD should be paired with the same UMD variant. When creating a workload container, depending on the target hardware, the UMD packages should be selected approriately.
+
+| KMD | UMD packages | Support notes |
+|:---- |:-------- |:------- |
+| `i915 upstream` | Distro Repository | For Integrated GPUs. Newer Linux kernels will introduce support for Arc, Flex or Max series. |
+| `i915 backport` | [Intel Repository](https://dgpu-docs.intel.com/driver/installation.html#install-steps) | Best for Arc, Flex and Max series. Untested for Integrated GPUs. |
+| `xe` | Not available yet | Experimental support for Arc, Flex and Max series. |
+
+Creating a workload that would support all the different KMDs is not currently possible. Below is a table that clarifies how each domain supports different KMDs.
+
+| Domain | i915 upstream | i915 backport | xe | Notes |
+|:---- |:-------- |:------- |:------- |:------- |
+| Compute | Default | [NEO_ENABLE_i915_PRELIM_DETECTION](https://github.com/intel/compute-runtime/blob/master/CMakeLists.txt#L498) | [NEO_ENABLE_XE_DRM_DETECTION](https://github.com/intel/compute-runtime/blob/master/CMakeLists.txt#L506) | All three KMDs can be supported at the same time. |
+| Media | Default | [ENABLE_PRODUCTION_KMD](https://github.com/intel/media-driver/blob/master/CMakeLists.txt#L58) | [ENABLE_NEW_KMD](https://github.com/intel/media-driver/blob/master/media_driver/cmake/linux/media_feature_flags_linux.cmake#L185) | Only one KMD can be enabled at a time. |
+| Graphics | Default | Default | [intel-xe-kmd](https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/meson_options.txt?ref_type=heads#L708) | All KMDs can be supported at the same time. |
```

Comment:
While this states Mesa supporting also backport (prelim) uAPI: https://dgpu-docs.intel.com/driver/kernel-driver-types.html

Mesa has worked only with upstream i915 & xe when I've last tested it. While it's a long time (may even be years) since I did that backport driver test, it's possible that Mesa is not being validated against out-of-tree KMDs, only upstream ones.

As to Xe driver support, Mesa is only one having that in a release, currently media & compute drivers have working support only in their repo master branches, and it working requires specific Xe driver tag / version, at least for now.

=> In addition to stating Xe KMD being experimental, UMD support for that could also be stated to be currently experimental (potentially breaking between versions).

Reply:
https://dgpu-docs.intel.com/driver/client/overview.html#client-install-options indicates that mesa could also be somehow changed. But I didn't see any compile/configuration flags in the mesa project.

I'll mark Xe UMD as "in development and experimental".